app/models/ability.rb

```diff
@@ -40,6 +43,12 @@ def initialize(user)
     label_abilities
   end
 
+  def guest_abilities
+    can %i[show download], Task, &:access_level_public?
+
+    can %i[show read], Comment
```

Comment:
Since comments are loaded asynchronously through JavaScript, it seems quite simple to change the URL and access some other comments. Can / should we include a check for the access level of the task as well? For example, we could decide that comments can only be read if the task is public.

Reply:
Yes, that is true. Actually the same problem exists with users that are logged in but are not allowed to access a certain task. For guest_abilities something like this works:
```
can %i[show read], Comment, task: {access_level: :public}
```

However this does not work for limiting access for users that are logged in because the access conditions in this case are more complex (is in same group?, is admin?, ...) and cannot be expressed as CanCanCan hash conditions. I found another solution though which authorizes access to comments depending on whether or not the user can access the comments task. 
https://github.com/CanCanCommunity/cancancan/blob/develop/docs/nested_resources.md

Reply:
> Actually the same problem exists with users that are logged in but are not allowed to access a certain task.

Indeed, I just noticed that as well.

> I found another solution though which authorizes access to comments depending on whether or not the user can access the comments task.

This looks perfect and is similar to the authorization concept we use in CodeOcean for nested resources. There, we don't have CanCanCan but another library, but nevertheless we are also checking the parent resource and then the actual object. 

Many thanks for these changes and the awesome test coverage you added! 👏